ja/reference/logical-replication.md

```diff
@@ -0,0 +1,316 @@
+---
+title: ロジカルレプリケーション
+---
+
+# ロジカルレプリケーション
+
+PGroongaは1.2.4からPostgreSQL組み込みのロジカルレプリケーションをサポートしています。ロジカルレプリケーションはPostgreSQL10以降が必要です。
+
+このドキュメントではPostgreSQL組み込みのロジカルレプリケーション機能をPGroonga用に設定する方法を説明します。多くの手順は通常のロジカルレプリケーションの設定手順です。いくつかPGroonga固有の手順があります。
+
+ロジカルレプリケーションはレプリケーション元とレプリケーション先で同じスキーマを持つ必要がありません。
+
+そのため、ここでは、レプリケーション先にのみインデックスを設定します。
+
+## 概要
+
+PostgreSQL組み込みのロジカルレプリケーション機能をPGroonga用に設定する手順は次の通りです。「[通常]」タグは通常のロジカルレプリケーションの手順であることを示しています。「[固有]」タグはPGroonga固有の手順であることを示しています。
+
+  1. [通常] パブリッシャーとサブスクライバーにPostgreSQLをインストールします。
+
+  2. [固有] サブスクライバーにPGroongaをインストールします。
+
+  3. [通常] パブリッシャーとサブスクライバーのPostgreSQLデータベースを初期化します。
+
+  4. [通常] パブリッシャーの`postgresql.conf`と`pg_hba.conf`にロジカルレプリケーションの設定を追加します。
+
+  5. [通常] パブリッシャーとサブスクライバーにテーブルを作成します。
+
+  6. [通常] パブリッシャーにパブリケーションを作成します。
+
+  7. [通常] サブスクライバーにサブスクリプションを作成します。
+
+  8. [固有] サブスクライバーにPGroongaのインデックスを作成します。
+
+  9. [通常] パブリッシャーにデータを挿入します。
+
+## 例で使う環境
+
+このドキュメントでは次の環境を使います。
+
+  * パブリッシャー:
+
+    * OS：CentOS 7
+
+    * IPアドレス: 172.16.0.1
+
+    * データベース名： `blog`
+
+    * レプリケーションユーザー名： `replicator`
+
+    * レプリケーションユーザーのパスワード：`passw0rd`
+
+  * サブスクライバー
+
+    * OS：CentOS 7
+
+    * IPアドレス: 172.16.0.2
+
+    * データベース名： `blog`
+
+このドキュメントではCentOS 7用のコマンドラインを書いています。もし、他のプラットフォームを使っている場合は自分でコマンドラインを調整してください。
+
+## [通常] パブリッシャーとサブスクライバーにPostgreSQLをインストールする。
+
+これは通常の手順です。
+
+パブリッシャーとサブスクライバーに PostgreSQL {{ site.latest_postgresql_version }} をインストールします。
+
+パブリッシャーとサブスクライバー
+
+```console
+% sudo -H yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-$(rpm -qf --queryformat="%{VERSION}" /etc/redhat-release)-$(rpm -qf --queryformat="%{ARCH}" /etc/redhat-release)/pgdg-redhat-repo-latest.noarch.rpm
+% sudo yum install postgresql{{ site.latest_postgresql_version }}
+% sudo yum install postgresql{{ site.latest_postgresql_version }}-server
+% sudo yum install postgresql{{ site.latest_postgresql_version }}-libs
+```
+
+[PostgreSQL: Linux downloads (CentOS)](https://www.postgresql.org/download/linux/redhat/) も参照してください。
+
+## [固有] サブスクライバーにPGroongaをインストールします。
+
+これはPGroonga固有の手順です。
+
+サブスクライバーにPGroongaをインストールします。
+
+サブスクライバー:
+
+```console
+% sudo -H yum install -y https://download.postgresql.org/pub/repos/yum/reporpms/EL-$(rpm -qf --queryformat="%{VERSION}" /etc/redhat-release)-$(rpm -qf --queryformat="%{ARCH}" /etc/redhat-release)/pgdg-redhat-repo-latest.noarch.rpm
+% sudo -H yum install -y https://packages.groonga.org/centos/groonga-release-latest.noarch.rpm
+% sudo -H yum install -y postgresql{{ site.latest_postgresql_version}}-pgroonga
+```
+
+## [通常] パブリッシャーとサブスクライバーのPostreSQLのデータベースを初期化します。
+
+これは通常の手順です。
+
+パブリッシャーとサブスクライバーのPostgreSQLのデータベースを初期化します。
+
+パブリッシャーとサブスクライバー
+
+```console
+% sudo /usr/pgsql-{{ site.latest_postgresql_version }}/bin/postgresql-{{ site.latest_postgresql_version }}-setup initdb
+% sudo systemctl enable postgresql-{{ site.latest_postgresql_version }}
+% sudo systemctl start postgresql-{{ site.latest_postgresql_version }}
```

Comment:
We can use `sudo systemctl enable --now ...` instead of `sudo systemctl enable ...` and `sudo systemctl start ...`.